File: NotesAppJv/app/src/main/java/com/henriquesbraga/notesappjv/activity/main/adapter/MainViewHolder.java
package com.henriquesbraga.notesappjv.activity.main.adapter;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.henriquesbraga.notesappjv.R;
import com.henriquesbraga.notesappjv.model.Note;

public class MainViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{

    TextView tv_title, tv_note, tv_date;
    CardView card_item;
    ItemClickListener itemClickListener;

    public MainViewHolder(@NonNull View itemView, ItemClickListener itemClickListener) {
        super(itemView);
        this.itemClickListener = itemClickListener;
    }

    public void bind(Note note){
        tv_title = itemView.findViewById(R.id.tv_title);
        tv_title.setText(note.getTitle());
        tv_note = itemView.findViewById(R.id.tv_note);
        tv_note.setText(note.getNote());
        tv_date = itemView.findViewById(R.id.tv_date);
        tv_date.setText(note.getDate());
        card_item = itemView.findViewById(R.id.card_item);
        card_item.setCardBackgroundColor(note.getColor());
        card_item.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        itemClickListener.onItemClick(view, getAdapterPosition());
    }
}

File: NotesAppKtSQ/app/src/main/java/com/henriquesbraga/notesappktsq/activity/main/adapter/MainViewHolder.kt
package com.henriquesbraga.notesappktsq.activity.main.adapter

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.henriquesbraga.notesappktsq.R
import com.henriquesbraga.notesappktsq.model.Note

class MainViewHolder(itemView: View, itemClickListener: ItemClickListener): RecyclerView.ViewHolder(itemView), View.OnClickListener {

    var itemClickListener = itemClickListener

    fun bind(note: Note){
        var tv_title: TextView = itemView.findViewById(R.id.tv_title)
        tv_title.text = note.title
        var tv_note = itemView.findViewById<TextView>(R.id.tv_note)
        tv_note.text = note.note
        var tv_date = itemView.findViewById<TextView>(R.id.tv_date)
        tv_date.text = note.date
        var card_item = itemView.findViewById<androidx.cardview.widget.CardView>(R.id.card_item)


        card_item.setBackgroundColor(note.color)

        card_item.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        itemClickListener?.onItemClick(v, adapterPosition)
    }

}
